stellar/deploy-contract.js

```diff
@@ -215,40 +216,33 @@ async function deploy(options, config, chain, contractName) {
     const wasmFile = await getWasmFile(options, contractName);
     const wasmHash = await uploadWasm(wasmFile, wallet, chain);
 
-    if (contractName === 'interchain_token' || contractName === 'token_manager') {
-        chain.contracts[contractName] = {
-            deployer: wallet.publicKey(),
-            wasmHash: serializeValue(wasmHash),
-        };
-    } else {
-        const initializeArgs = await getInitializeArgs(config, chain, contractName, wallet, options);
-        const serializedArgs = Object.fromEntries(
-            Object.entries(initializeArgs).map(([key, value]) => [key, serializeValue(scValToNative(value))]),
-        );
-        const operation = Operation.createCustomContract({
-            wasmHash,
-            address: Address.fromString(wallet.publicKey()),
-            // requires that initializeArgs returns the parameters in the appropriate order
-            constructorArgs: Object.values(initializeArgs),
-        });
-        printInfo('Initializing contract with args', JSON.stringify(serializedArgs, null, 2));
-
-        const deployResponse = await broadcast(operation, wallet, chain, 'Initialized contract', options);
-        const contractAddress = StrKey.encodeContract(Address.fromScAddress(deployResponse.address()).toBuffer());
-
-        validateParameters({
-            isValidStellarAddress: { contractAddress },
-        });
-
-        printInfo('Contract initialized at address', contractAddress);
-
-        chain.contracts[contractName] = {
-            address: contractAddress,
-            deployer: wallet.publicKey(),
-            wasmHash: serializeValue(wasmHash),
-            initializeArgs: serializedArgs,
-        };
-    }
+    const initializeArgs = await getInitializeArgs(config, chain, contractName, wallet, options);
+    const serializedArgs = Object.fromEntries(
+        Object.entries(initializeArgs).map(([key, value]) => [key, serializeValue(scValToNative(value))]),
+    );
+    const operation = Operation.createCustomContract({
+        wasmHash,
+        address: Address.fromString(wallet.publicKey()),
+        // requires that initializeArgs returns the parameters in the appropriate order
+        constructorArgs: Object.values(initializeArgs),
+    });
+    printInfo('Initializing contract with args', JSON.stringify(serializedArgs, null, 2));
+
+    const deployResponse = await broadcast(operation, wallet, chain, 'Initialized contract', options);
+    const contractAddress = StrKey.encodeContract(Address.fromScAddress(deployResponse.address()).toBuffer());
```

Comment:
```suggestion
    const contractAddress = Address.fromScAddress(deployResponse.address()).toString();
```
is there a direct string conversion?

Reply:
good catch. updated